submission/submission-webapp/src/main/kotlin/ac/uk/ebi/biostd/submission/web/handlers/SubmitWebHandler.kt

```diff
@@ -10,48 +10,80 @@ import ac.uk.ebi.biostd.submission.domain.service.SubmissionService
-        val source = sources(request.user, sub, request.files)
+        val source = sources(request.submitter, sub, request.files)
         val submission = withAttributes(submission(request.submission, request.format, source), request.attrs)
-        return submissionService.submit(SubmissionRequest(submission, request.user, source, PAGE_TAB, request.fileMode))
-    }
-
-    fun refreshSubmission(request: RefreshWebRequest): Submission {
-        val submission = submissionService.getSubmission(request.accNo).toSimpleSubmission()
-        val source = sources(request.user, submission)
-        return submissionService.submit(SubmissionRequest(submission, request.user, source, PAGE_TAB, FileMode.MOVE))
+        return submissionService.submit(Request(
+            submission = submission,
+            submitter = request.submitter,
+            onBehalfUser = request.onBehalfRequest?.let { getOnBehalfUser(it) },
+            method = PAGE_TAB,
+            sources = source,
+            mode = request.fileMode
+        ))
     }
 
     fun submit(request: FileSubmitWebRequest): Submission {
         val sub = serializationService.deserializeSubmission(request.submission)
-        val source = sources(request.user, sub, request.files.plus(request.submission))
+        val source = sources(request.submitter, sub, request.files.plus(request.submission))
         val submission = withAttributes(submission(request.submission, source), request.attrs)
-        userFilesService.uploadFile(request.user, DIRECT_UPLOAD_PATH, request.submission)
-        return submissionService.submit(SubmissionRequest(submission, request.user, source, FILE, request.fileMode))
+        userFilesService.uploadFile(request.submitter, DIRECT_UPLOAD_PATH, request.submission)
+        return submissionService.submit(Request(
+            submission = submission,
+            submitter = request.submitter,
+            onBehalfUser = request.onBehalfRequest?.let { getOnBehalfUser(it) },
+            sources = source,
+            method = FILE,
+            mode = request.fileMode
+        ))
     }
 
-    private fun sources(user: SecurityUser, submission: Submission, files: List<File> = emptyList()): FilesSource {
-        val request = RequestSources(user, files, submission.rootPath, subFolder(submission.accNo))
-        return sourceGenerator.submissionSources(request)
+    fun refreshSubmission(request: RefreshWebRequest): Submission {
+        val submission = submissionService.getSubmission(request.accNo).toSimpleSubmission()
+        val source = sources(request.user, submission, emptyList())
+        return submissionService.submit(Request(
+            submission = submission,
+            submitter = request.user,
+            sources = source,
+            method = PAGE_TAB,
+            mode = FileMode.MOVE
+        ))
+    }
+
+    private fun getOnBehalfUser(request: OnBehalfRequest): SecurityUser =
```

Comment:
Are we sure about creating a user if it doesn't exist? I would expect `onBehalf` functionality for only work with existing users. We're probably mixing two things here.

Reply:
that is needed and that is current system behavior,  unified guys need user creation as they submit with the user on behalf of another and owner need to be new user.